tests/simulators/carla/test_actions.py

```diff
@@ -0,0 +1,102 @@
+import os
+import socket
+import subprocess
+
+import pytest
+
+try:
+    import carla
+except ModuleNotFoundError:
+    pytest.skip("carla package not installed", allow_module_level=True)
+
+from tests.utils import compileScenic, sampleScene
+
+
+def checkCarlaPath():
+    CARLA_ROOT = os.environ.get("CARLA_ROOT")
+    if not CARLA_ROOT:
+        pytest.skip("CARLA_ROOT env variable not set.")
+    return CARLA_ROOT
+
+
+def launchCarlaServer():
+    CARLA_ROOT = checkCarlaPath()
+    subprocess.Popen(f"bash {CARLA_ROOT}/CarlaUE4.sh -RenderOffScreen", shell=True)
```

Comment:
I'd worry about leaving CARLA open indefinitely (we might test other simulators and only have the resources to run one at a time). I'd suggest having `getCarlaSimulator` both start and stop CARLA, and give it [package scope](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session) so that the fixture will be shut down after all the CARLA tests are done.

Reply:
Ack! Added cleanup for the carla server subprocess along with adding the `package` scope for the fixture!